### ISSUE TITLE
Pre-Commit support to improve code quality

### ISSUE THREAD

**Issue by hofbi** [pytest-dev/pyfakefs#733]
[pre-commit](https://pre-commit.com/) is a pretty nice tool to manage other tools such as code formatters and linters used to improve the code quality. It is used by many other projects such as [pytest](https://github.com/pytest-dev/pytest) itself.

It would be really great to add this to this repo as well.

**Comment by mrbean-bremen** [comment]
Thanks, that sounds useful! I will certainly try this out. As I understand, this would also replace the `flake8` and `mypy` runs we currently do in the test workflow.

**Comment by hofbi** [comment]
Exactly. Or at least you don't need to call them directly but use pre-commit to run it.

If you then want to go even one step further, you could also enable the pre-commit CI which is free for open source projects.

**Comment by mrbean-bremen** [comment]
Works fine - thanks for the idea! 
I will probably add `black` sometime later. Currently auto-black changes too much in the code and creates multiple flake8 issues along the line (long lines).

**Comment by hofbi** [comment]
Nice, thank you.

You can also check my flake8 config. It disables two checks that cause conflicts between black and flake8: https://github.com/hofbi/teaching-tools/blob/master/.flake8

**Comment by mrbean-bremen** [comment]
Thank you, I had considered this. Actually I had the line length set to 120 in my projects first, but changed it to the pep8 version (e.g. 79) after some discussions in another project. I'm a bit undecided here: while I'm personally perfectly fine with 120 characters (especially considering the usage of long names), it seems to be a bit of a controversal issue, and I can live with both. 
Though if this is the standard setting in `black`, and as `black` is nowadays considered the standard by many, I may change this...

**Comment by hofbi** [comment]
Sounds good. I don't have a strong opinion on the code format as long as a formatter does it consistent across the codebase
